VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -2,32 +2,52 @@
 
 import android.content.res.Resources;
 import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
 import android.text.Html;
-import android.text.Spanned;
-import android.text.TextUtils;
 import android.text.format.DateUtils;
 
 import org.edx.mobile.R;
+import org.edx.mobile.util.ResourceUtil;
+
+import java.util.HashMap;
 
 public abstract class DiscussionTextUtils {
     private DiscussionTextUtils() {
     }
 
-    public static String getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources) {
-        String text = authorData.getAuthor() + " " + DateUtils.getRelativeTimeSpanString(
+    public static CharSequence getAuthorAttributionText(@NonNull DiscussionThread discussionThread, @NonNull Resources resources) {
+        return getAuthorAttributionText(discussionThread, resources,
+                discussionThread.getType() == DiscussionThread.ThreadType.DISCUSSION
+                        ? R.string.discussion_thread_attribution
+                        : R.string.question_thread_attribution);
+    }
+
+    public static CharSequence getAuthorAttributionText(@NonNull DiscussionComment discussionComment, @NonNull Resources resources) {
+        return getAuthorAttributionText(discussionComment, resources, R.string.comment_attribution);
+    }
+
+    private static CharSequence getAuthorAttributionText(@NonNull final IAuthorData authorData, @NonNull final Resources resources, @StringRes int stringRes) {
```

Comment:
There seems to be some discrepancy among the `final` modifier in the parameters; only the first two parameters are `final`, and only in this last method.